src/webui/webapplication.cpp

```diff
@@ -212,12 +208,10 @@ void WebApplication::translateDocument(QString &data)
             const QString sourceText = regexMatch.captured(1);
             const QString context = regexMatch.captured(3);
 
-            QString translation = sourceText;
-            if (isTranslationNeeded) {
-                const QString loadedText = m_translator.translate(context.toUtf8().constData(), sourceText.toUtf8().constData(), nullptr, 1);
-                if (!loadedText.isEmpty())
-                    translation = loadedText;
-            }
+            const QString loadedText = m_translator.translate(context.toUtf8().constData(), sourceText.toUtf8().constData());
```

Comment:
Shouldn't you optimize it? You can skip call to `QTranslator::translate()` in case of missing translation file for selected language.

Reply:
Indeed! added back.